packages/plugin-vue-jsx/src/index.ts

```diff
@@ -85,197 +86,215 @@ function vueJsxPlugin(options: Options = {}): Plugin {
       root = config.root
     },
 
-    resolveId(id) {
-      if (id === ssrRegisterHelperId) {
-        return id
-      }
+    resolveId: {
+      filter: { id: exactRegex(ssrRegisterHelperId) },
+      handler(id) {
+        if (id === ssrRegisterHelperId) {
```

Comment:
We still need to keep the `if` statements to keep the compat for Vite older than v6.3.